_pytest/python.py

```diff
@@ -1115,7 +1115,10 @@ def _idval(val, argname, idx, idfn):
     elif isinstance(val, (float, int, str, bool, NoneType)):
         return str(val)
     elif isinstance(val, REGEX_TYPE):
-        return val.pattern
+        if isinstance(val.pattern, str):
```

Comment:
While this works, I think a better approach here would be to reuse the effort made by `_escape_bytes` to handle `bytes` as ascii-escaped values instead of trying to decode from `utf-8`:

```python
elif isinstance(val, REGEX_TYPE):
        return _escape_bytes(val.pattern) if isinstance(val.pattern, bytes) else val.pattern
```